### PR TITLE
Split up plotting examples into single files

### PR DESCRIPTION
**Pull request by wilzbach**
@9il I think I found a good way to restructure the plotting code, s.t. it's more user-friendly
## Compile & run a single plot

I put every plotting file into it's its own file that's runnable & executable. This should make it a lot easier to get an overview as one doesn't need to change a 700 LoC file. Of course all options can be modified directly by the user in a single example plot file.
They all use the new single file dub syntax and can be run like this:

```
./examples/flex_plot/test_normal.d
```
## Compile & run all plots

Also supports the dub single file syntax:

```
./examples/flex_plot.d
```

The common plotter also provides convenience CLI options that can be used to manipulate some plotting options on the CLI manually or only select a subset of plots.
If wanted I can move the CLI parsing also to common (it could be useful, but the single files are pretty small)

There are still some minor things to be done, but I wanted to hear your opinion on this idea.

### CODE REVIEW

File: examples/flex_plot/test1.d [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env dub
+/+ dub.sdl:
```
Comment (9il) [inline examples/flex_plot/test1.d:2]: Should be json, see #258

**Comment by wilzbach** [comment]
@9il seems like PyD doesn't support Python3.5 - see https://github.com/ariovistus/pyd/issues/50 for details :/
My workaround is to patch the `dub.json` from `appveyor.yml` - do you have a better idea what we could do for Windows users? (I just hope that nobody in their sane mind tries to use Windows)

**Comment by wilzbach** [comment]
@9il this is finally passing on CircleCi and AppVeyor :tada: 
Can I go ahead, merge and improve upon (e.g. the `sdl` to `json` thing is a really minor thing)?

**Comment by wilzbach** [comment]
Btw for the protocol, building Python 3.5 by source on CircleCi is needed as otherwise we get

```
/usr/bin/ld: cannot find -lpython3.5m
```

**Comment by wilzbach** [comment]
> Btw for the protocol, building Python 3.5 by source on CircleCi is needed

Thanks to the caching at CircleCi it's adds only one minute extra to the build time.

**Pull request merged**